cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/timeseries/FactScanner.java

```diff
@@ -45,15 +46,19 @@
   // Use an internal iterator to avoid leaking AbstractIterator methods to outside.
   private final Iterator<FactScanResult> internalIterator;
 
+  // set of measureNames - useful to process measures that are requested while scanning.
+  private final ImmutableSet<String> measureNames;
+
   /**
    * Construct a FactScanner. Should only be called by FactTable.
    */
-  FactScanner(Scanner scanner, FactCodec codec, long startTs, long endTs) {
+  FactScanner(Scanner scanner, FactCodec codec, long startTs, long endTs, List<String> measureNames) {
     this.scanner = scanner;
     this.codec = codec;
     this.internalIterator = createIterator();
     this.startTs = startTs;
     this.endTs = endTs;
+    this.measureNames = (measureNames == null) ? ImmutableSet.<String>of() : ImmutableSet.copyOf(measureNames);
```

Comment:
I guess that is related to my previous question about empty measure names.. So based on this, empty means "all metrics"?